backend/middlewares/sessionManager.js

```diff
@@ -27,6 +27,12 @@ module.exports = async (req, res, next) => {
 		return;
 	}
 
+	if ((req.headers.referer && !req.headers.referer.match(`^${config.public_protocol}://${config.cookie_domain}`)) ||
+		(!req.headers.referer && !config.cookie.skipMissingReferer)) {
+		req[C.REQ_REPO].logger.logError(`express-session internal error: referer (${req.headers.referer}) and cookie (${config.cookie_domain}) domain mismatch`);
+		return;
```

Comment:
If we're not going next(), we need to respond to the request here. at the moment if it enters this condition the request never returns.